src/NetworkName.js

```diff
@@ -1,60 +1,134 @@
     }
-}
 
-/**
- * @param {string} ledgerId
- * @returns {string}
- */
-export function _ledgerIdToNetworkName(ledgerId) {
-    switch (ledgerId) {
-        case "0":
-            return NetworkName.Mainnet;
-        case "1":
-            return NetworkName.Testnet;
-        case "2":
-            return NetworkName.Previewnet;
-        default:
-            throw new Error(`unrecognized ledgerId, expected "0", "1", or "2"`);
+    /**
+     * @param {Uint8Array} bytes
+     * @returns {NetworkName}
+     */
+    static fromBytes(bytes) {
+        return NetworkName.fromString(NetworkName.toName(bytes.toString()));
+    }
+
+    /**
+     * @returns {string}
+     */
+    toString() {
+        return typeof this._networkName == "string"
+            ? this._networkName
+            : this._networkName.toString();
+    }
+
+    /**
+     * @param {string} networkName
+     * @returns {NetworkName}
+     */
+    static fromString(networkName) {
+        switch (networkName) {
+            case NETNAMES[0]:
+            case "0":
+                return MAINNET;
+            case NETNAMES[1]:
+            case "1":
+                return TESTNET;
+            case NETNAMES[2]:
+            case "2":
+                return PREVIEWNET;
+            case NETNAMES[3]:
+            case "3":
+                return OTHER;
+            default:
+                throw new Error(
+                    DEFAULT_ERROR.concat("fromString: networkName = ").concat(
+                        networkName
+                    )
+                );
+        }
+    }
+
+    /**
+     * @param {LedgerId | number | string} ledgerId
+     * @returns {string}
+     */
+    static toName(ledgerId) {
+        ledgerId =
+            ledgerId instanceof LedgerId ? ledgerId.toString() : ledgerId;
+
+        if (NETNAMES.includes(ledgerId.toString())) {
+            //if LedgerId was constructed with a network name, return name
+            return ledgerId.toString();
+        }
+
+        switch (ledgerId) {
+            case 0:
+            case "0":
+                return NETNAMES[0];
+            case 1:
+            case "1":
+                return NETNAMES[1];
+            case 2:
+            case "2":
+                return NETNAMES[2];
+            default:
+                return NETNAMES[3];
+        }
+    }
+
+    /**
+     * @param {string} networkName
+     * @returns {number}
+     */
+    static toId(networkName) {
+        switch (networkName) {
+            case NETNAMES[0]:
+                return 0;
+            case NETNAMES[1]:
+                return 1;
+            case NETNAMES[2]:
+                return 2;
+            default:
+                return 3;
+        }
     }
 }
 
-export default NetworkName;
+const NETNAMES = ["mainnet", "testnet", "previewnet", "other"];
+
+const MAINNET = new NetworkName("mainnet");
+
+const TESTNET = new NetworkName("testnet");
+
+const PREVIEWNET = new NetworkName("previewnet");
```

Comment:
The comment made before was only for `NETNAMES`, you'd still want to declare this and the other `*NET` types as `NetworkName.MAINNET = new NetworkName("mainnet");`